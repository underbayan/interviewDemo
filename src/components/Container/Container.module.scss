.containerStyle {
  padding-top: 120px;
  padding-bottom: 120px;
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 100vh;
  box-sizing: border-box;
  flex-direction: column;
}