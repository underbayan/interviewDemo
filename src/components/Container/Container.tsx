import { memo, ReactNode } from 'react';
import { containerStyle } from "./Container.module.scss"

interface ContainerProps {
  children: ReactNode;
}

export const Container = memo(({ children }: ContainerProps) => {
  return (
    <main className={containerStyle + " p-t-sm-1 p-t-md-2"}>
      {children}
    </main>
  )
})
