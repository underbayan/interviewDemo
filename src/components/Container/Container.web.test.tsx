import { render } from '@testing-library/react';
import { test, expect } from 'vitest';
import { Container } from './Container';
test('Container renders correctly', () => {
  const { container } = render(<Container>Click me</Container>);
  expect(container).toMatchSnapshot();
});
