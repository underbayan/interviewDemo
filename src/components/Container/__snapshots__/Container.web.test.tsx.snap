// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container renders correctly 1`] = `
<div>
  <main
    class="undefined p-t-sm-1 p-t-md-2"
  >
    Click me
  </main>
</div>
`;
